### PR TITLE
Suppress assertion failure when using CComMultiThreadModel.

### PR DESCRIPTION
**Pull request by patwonder**
Using CComMultiThreadModel for the sink and protocol objects resolves a random crash problem in higher versions of IE(IE10 & 11). See http://social.msdn.microsoft.com/Forums/ie/en-US/f2a63a7e-dcd0-4d76-9824-94c8b81a3e9c/app-sdk-crash-in-iinternetprotocolimplunlockrequest?forum=ieextensiondevelopment for the detailed discussion.

Unfortunately, this version of PassthroughAPP generates a runtime assertion failure when using CComMultiThreadModel for the sink and protocol objects:

``` C++
In ATL::CComSafeDeleteCriticalSection::Lock()
    HRESULT Lock()
    {
        // CComSafeDeleteCriticalSection::Init or CComAutoDeleteCriticalSection::Init
        // not called or failed.
        // m_critsec member of CComObjectRootEx is now of type
        // CComAutoDeleteCriticalSection. It has to be initialized
        // by calling CComObjectRootEx::_AtlInitialConstruct
        ATLASSUME(m_bInitialized);
        return CComCriticalSection::Lock();
    }
```

The reason is that calling _AtlInitialConstruct is needed in ATL version 8 (or higher) but is not called in the FinalConstruct function of the protocol and sink objects. This patch adds these calls.

### CODE REVIEW

File: SinkPolicy.inl [hunk 1]
```diff
@@ -132,7 +132,16 @@ inline HRESULT CComPolyObjectSharedRef<Contained>::FinalConstruct()
 	InternalAddRef();
 	CComObjectRootEx<Contained::_ThreadModel::ThreadModelNoCS>::
 		FinalConstruct();
-	HRESULT hr = m_contained.FinalConstruct();
+	HRESULT hr;
+#if _ATL_VER >= 0x800
+	hr = m_contained._AtlInitialConstruct();  // gears - added by andreip
```
Comment (matthewgertner) [inline SinkPolicy.inl]: What exactly do these comments mean? Do they make sense in this context or are they something specific to Gears?

**Comment by matthewgertner** [comment]
Also should we change the default thread model for the protocol and sink to `CComMultiThreadModel`?

File: SinkPolicy.inl (hunk 1 above)
Comment (patwonder) [inline SinkPolicy.inl]: @matthewgertner They come from the original source of Google Gears. I just haven't removed them.

**Comment by patwonder** [comment]
@matthewgertner I would suggest changing them to use `CComMultiThreadModel`, because
1. It's unlikely that this change will break existing code.
2. It's always safer to use the multi-thread model, regardless of IE's version.
3. Performance impact should be negligible.

**Comment by matthewgertner** [comment]
Do you mind updating your pull request to remove the Gears comments and change the threading model?

**Comment by patwonder** [comment]
@matthewgertner Done.

**Comment by matthewgertner** [comment]
Thanks, @patwonder!

**Pull request merged**